hal/src/main/native/sim/PowerDistribution.cpp

```diff
@@ -147,11 +147,23 @@ void HAL_GetPowerDistributionAllChannelCurrents(
 }
 double HAL_GetPowerDistributionTotalCurrent(HAL_PowerDistributionHandle handle,
                                             int32_t* status) {
-  return 0.0;
+  auto module = hal::can::GetCANModuleFromHandle(handle, status);
+  if (*status != 0) {
+    return 0.0;
+  }
+
+  double total = 0.0;
+  auto& data = SimPowerDistributionData[module];
+  for (int i = 0; i < kNumPDSimChannels; i++) {
```

Comment:
This might be problematic if the array size might change (for simulating channel count)

Reply:
kNumPDSimChannels drives the array size. The other functions depend on that constant as well.